mlflow/_promptlab.py

```diff
@@ -0,0 +1,154 @@
+# loader/gateway_loader_module.py
+import os
+from string import Template
+from typing import List
+
+import pandas as pd
+import yaml
+
+import mlflow.gateway
+from mlflow import pyfunc
+from mlflow.models import Model
+from mlflow.models.model import MLMODEL_FILE_NAME, Model
+from mlflow.models.utils import _save_example
+from mlflow.pyfunc.model import PythonModel
+from mlflow.utils.environment import (
+    _CONDA_ENV_FILE_NAME,
+    _CONSTRAINTS_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _process_conda_env,
+    _process_pip_requirements,
+    _PythonEnv,
+    _validate_env_arguments,
+)
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.model_utils import (
+    _get_flavor_configuration,
+    _validate_and_copy_code_paths,
+    _validate_and_prepare_target_save_path,
+)
+
+
+class _PromptlabModel(PythonModel):
+    def __init__(self, prompt_template, prompt_parameters, model_parameters, model_route):
+        self.prompt_parameters = prompt_parameters
+        self.model_parameters = model_parameters
+        self.model_route = model_route
+        self.prompt_template = Template(prompt_template)
+
+    def predict(self, inputs: pd.DataFrame) -> List[str]:
+        results = []
+        for idx in inputs.index:
+            prompt_parameters_as_dict = {
+                param.key: inputs["{param.key}"][idx] for param in self.prompt_parameters
+            }
+            prompt = self.prompt_template.substitute(prompt_parameters_as_dict)
+            model_parameters_as_dict = {param.key: param.value for param in self.model_parameters}
+            result = mlflow.gateway.query(
+                route=self.model_route,
+                data={
+                    {
+                        "prompt": prompt,
+                    }.update(model_parameters_as_dict),
+                },
+            )
```

Comment:
/ai what value is passed to the `data` argument in this function call?

Reply:
`dict.update` returns `None`. `{...}` is a set literal. `{None,}` is passed to `data`:

```python
prompt = 1
model_parameters_as_dict = {}
data = {
    {
        "prompt": prompt,
    }.update(model_parameters_as_dict),
}
print(data)
# {None}
```